pkg/suggestion/v1alpha2/bayesian_service.py

```diff
@@ -0,0 +1,212 @@
+                          extra={"experiment_name": request.experiment_name})
+        self.logger.debug("upperbound: %r", upperbound,
+                          extra={"experiment_name": request.experiment_name})
+        alg = BOAlgorithm(
+            dim=algo_manager.dim,
+            N=int(service_params["N"]),
+            lowerbound=lowerbound,
+            upperbound=upperbound,
+            X_train=algo_manager.X_train,
+            y_train=algo_manager.y_train,
+            mode=service_params["mode"],
+            trade_off=service_params["trade_off"],
+            # todo: support length_scale with array type
+            length_scale=service_params["length_scale"],
+            noise=service_params["noise"],
+            nu=service_params["nu"],
+            kernel_type=service_params["kernel_type"],
+            n_estimators=service_params["n_estimators"],
+            max_features=service_params["max_features"],
+            model_type=service_params["model_type"],
+            logger=self.logger,
+        )
+        trials = []
+        x_next_list = alg.get_suggestion(request.request_number)
+        for x_next in x_next_list:
+            x_next = x_next.squeeze()
+            self.logger.debug("xnext: %r ", x_next, extra={
+                              "experiment_name": request.experiment_name})
+            x_next = algo_manager.parse_x_next(x_next)
+            x_next = algo_manager.convert_to_dict(x_next)
+            trials.append(api_pb2.Trial(
+                spec=api_pb2.TrialSpec(
+                    experiment_name=request.experiment_name,
+                    parameter_assignments=api_pb2.TrialSpec.ParameterAssignments(
+                        assignments=[
+                            api_pb2.ParameterAssignment(
+                                name=x["name"],
+                                value=str(x["value"]),
+                            ) for x in x_next
+                        ]
+                    )
+                )
+            ))
+        return api_pb2.GetSuggestionsReply(
+            trials=trials
+        )
+
+    def getEvalHistory(self, experiment_name, obj_name, burn_in):
+        worker_hist = []
+        x_train = []
+        y_train = []
+        channel = grpc.beta.implementations.insecure_channel(
+            self.manager_addr, self.manager_port)
+        with api_pb2.beta_create_Manager_stub(channel) as client:
+            trialsrep = client.GetTrialList(api_pb2.GetTrialListRequest(
+                experiment_name=experiment_name
+            ))
+            for t in trialsrep.trials:
+                if t.status.condition == 2:
+                    gwfrep = client.GetObservationLog(
+                        api_pb2.GetObservationLogRequest(
+                            trial_name=t.name,
+                            metric_name=obj_name))
+                    w = gwfrep.observation_log
+                    for ml in w.metrics_logs:
+                        if ml.name == obj_name:
```

Comment:
I don't think, that we need condition here, we select values only for `metric_name=obj_name`